src/runner/browser-set.js

```diff
@@ -120,11 +121,9 @@ export default class BrowserSet extends EventEmitter {
     }
 
     async dispose () {
-        // FIXME: make a shallow copy of `this.connections`, because it
-        // will be modified inside `releaseConnection`.
-        // Not the most clear piece of code here. I hope we can do
-        // better. But I'm out of ideas for now.
-        this.connections.slice().forEach(bc => this.releaseConnection(bc));
+        // NOTE: When taskPromise is cancelled it is removed from pendingTaskPromises array, that leads to shifting indexes
+        // to the beginning. So in order to iterate the array we must copy it, or perform iteration from the end to beginning.
```

Comment:
Can we keep original formatting of the comment? It's hard to read, when it's that long.